src/core/createGetLinkElementDetails.js

```diff
@@ -0,0 +1,50 @@
+// eslint-disable-next-line import/no-named-as-default
+import createGetLinkDetails from "../components/ActivityCollector/createGetLinkDetails";
+
+export default (window, config, logger) => {
+  const { onBeforeLinkClickSent } = config;
+  return element => {
+    if (!element) {
+      return undefined;
+    }
+    const getLinkDetails = createGetLinkDetails(window, config);
```

Comment:
Pass this into the function directly and create it outside of this function. i.e.
```
const getLinkDetails = createGetLinkDetails({ window, config });
const getLinkElementDetails = createGetLinkElementDetails({ getLinkDetails, onBeforeLinkClickSend, logger });
```
In general, pass the object into the function instead of importing the create function. Avoiding the import function lets us test each file on its own. The exception to this guideline is to import simple util functions directly.

I notice that you do this in a few places. We want to separate the code we use to do dependency injection (which is only tested with functional tests), and the code we implement business logic (which can be tested with unit tests). When you import a file directly you can no longer write a unit test without also testing the code in the imported file. It's better to inject the function so that you can stub it out in the unit test.